chaco/legend.py

```diff
@@ -434,35 +461,58 @@ def get_label_at(self, x, y):
             return None
 
     def _do_layout(self):
-        if self.component is not None or len(self._cached_labels) == 0 or \
-                self._cached_label_sizes is None or len(self._cached_label_names) == 0:
+        if (
+            self.component is not None
+            or len(self._cached_labels) == 0
+            or self._cached_label_sizes is None
+            or len(self._cached_label_names) == 0
+        ):
             width, height = self.get_preferred_size()
             self.outer_bounds = [width, height]
 
     def _create_label(self, text):
-        """ Returns a new Label instance for the given text.  Subclasses can
+        """Returns a new Label instance for the given text.  Subclasses can
         override this method to customize the creation of labels.
         """
-        return Label(text=text, font=self.font, margin=0, color=self.color_,
-                     bgcolor="transparent", border_width=0)
+        return Label(
+            text=text,
+            font=self.font,
+            margin=0,
+            color=self.color_,
+            bgcolor="transparent",
+            border_width=0,
+        )
 
     def _composite_icon_renderer_default(self):
         return CompositeIconRenderer()
 
-    #-- trait handlers --------------------------------------------------------
+    # -- trait handlers --------------------------------------------------------
     def _anytrait_changed(self, name, old, new):
-        if name in ("font", "border_padding", "padding", "line_spacing",
-                    "icon_bounds", "icon_spacing", "labels", "plots",
-                    "plots_items", "labels_items", "border_width", "align",
-                    "position", "position_items", "bounds", "bounds_items",
-                    "label_at_top"):
+        if name in (
+            "font",
+            "border_padding",
+            "padding",
+            "line_spacing",
+            "icon_bounds",
+            "icon_spacing",
+            "labels",
+            "plots",
+            "plots_items",
+            "labels_items",
+            "border_width",
+            "align",
+            "position",
+            "position_items",
+            "bounds",
+            "bounds_items",
+            "label_at_top",
+        ):
             self._layout_needed = True
```

Comment:
here's another fun little trait which seems to have a trait change handler hooked up somewhere which redraws or something when it is set.